rudder-agent/SOURCES/signature.sh

```diff
@@ -1,54 +1,9 @@
 #!/bin/sh
 
-# To allow Rudder to provide its own version of openssl
-PATH=/opt/rudder/bin:$PATH
-export PATH
-
-if type openssl 2>/dev/null >/dev/null
-then
-  :
-else
-  echo "ERROR: openssl binary is missing !"
-  exit 1
-fi
-
-# md4 md5 sha sha1 sha224 sha256 sha384 sha512 whirlpool
-# The oldest openssl we support is 0.9.8 and it supports sha512
-HASH=sha512
-
-# the file to sign
-FILE="$1"
-if [ ! -f "${FILE}" ]
-then
-  echo "ERROR: Cannot sign: The file ${FILE} doesn't exist"
-  exit 2
-fi
-
-# the key to use for signature
-PRIVKEY=/var/rudder/cfengine-community/ppkeys/localhost.priv
-
-# cfengine  passphrase
-PASSPHRASE="Cfengine passphrase"
-
-# Create signature
-SIGNATURE=`openssl dgst -passin "pass:${PASSPHRASE}" -${HASH} -hex -sign "${PRIVKEY}" < "${FILE}" | sed -e 's/.*= //'`
-
-# Hostname (informative only, it can be slightly different from the rudder detected one)
-HOSTNAME=`hostname`
-
-# Private key modification date
-KEYDATE=`stat -c %y ${PRIVKEY}`
-
-# Public key identifier (last 4 bytes of the modulus)
-KEYID=`openssl rsa -passin "pass:${PASSPHRASE}" -in "${PRIVKEY}" -noout -modulus | sed 's/.*\(........\)$/\1/'`
-
-# Create a signature FILE
-cat > "${FILE}.sign" <<EOF
-header=rudder-signature-v1
-algorithm=${HASH}
-digest=${SIGNATURE}
-hostname=${HOSTNAME}
-keydate=${KEYDATE}
-keyid=${KEYID}
-EOF
+echo "WARNING: This command is deprecated and will be removed in a future version of Rudder."
+echo "WARNING: Use $(dirname $0)/rudder-sign instead."
+echo ""
+echo "Redirecting to $(dirname $0)/rudder-sign..."
+echo ""
 
+$(dirname $0)/rudder-sign $*
```

Comment:
This script is called for the inventory technique, you should change the name there too.